src/librbd/ImageCtx.h

```diff
@@ -154,6 +192,9 @@ namespace librbd {
     xlist<operation::ResizeRequest<ImageCtx>*> resize_reqs;
 
     io::ImageRequestWQ *io_work_queue;
+    ImageQosStatusSingleton* qos_status;
+    SafeTimer *m_timer;
```

Comment:
@dillaman we have removed the additional timer added previously, now "m_timer" directly reference the SafetimerSingleton. we will address other comments you just mentioned.